cli/internal/cloudcmd/iam.go

```diff
@@ -201,23 +200,51 @@ func (c *IAMCreator) createAWS(ctx context.Context, cl tfIAMClient, opts *IAMCon
 	}
 
 	if err := cl.PrepareWorkspace(path.Join("terraform", "iam", strings.ToLower(cloudprovider.AWS.String())), &vars); err != nil {
-		return iamid.File{}, err
+		return IAMOutput{}, err
 	}
 
 	iamOutput, err := cl.ApplyIAMConfig(ctx, cloudprovider.AWS, opts.TFLogLevel)
 	if err != nil {
-		return iamid.File{}, err
+		return IAMOutput{}, err
 	}
 
-	return iamid.File{
+	return IAMOutput{
 		CloudProvider: cloudprovider.AWS,
-		AWSOutput: iamid.AWSFile{
+		AWSOutput: AWSIAMOutput{
 			WorkerNodeInstanceProfile:   iamOutput.AWS.WorkerNodeInstanceProfile,
 			ControlPlaneInstanceProfile: iamOutput.AWS.ControlPlaneInstanceProfile,
 		},
 	}, nil
 }
 
+// IAMOutput is the output of creating a new IAM profile.
+type IAMOutput struct {
+	// CloudProvider is the cloud provider of the cluster.
+	CloudProvider cloudprovider.Provider `json:"cloudprovider,omitempty"`
+
+	GCPOutput   GCPIAMOutput   `json:"gcpOutput,omitempty"`
```

Comment:
should we make the fields optional through a ptr?

Reply:
They weren't before. This struct also is never written to disk, so making fields optional doesn't change much, but requires us to make additional safety checks (the json tags also don't really matter here).